test/03_USM.test.js

```diff
@@ -255,186 +212,369 @@ contract('USM', (accounts) => {
+            shouldEqual(minFumBuyPrice4, targetMinFumBuyPrice4)
+
+            // Now move forward a few days, and check that minFumBuyPrice decays by the appropriate factor:
+            const block0 = await web3.eth.getBlockNumber()
             const t0 = (await web3.eth.getBlock(block0)).timestamp
-            const timeDelay = 3 * MINUTE
+            const timeDelay = 3 * DAY
             await timeMachine.advanceTimeAndBlock(timeDelay)
-            const block1 = (await web3.eth.getBlockNumber())
+            const block1 = await web3.eth.getBlockNumber()
             const t1 = (await web3.eth.getBlock(block1)).timestamp
             shouldEqual(t1, t0 + timeDelay)
 
-            const mintBurnAdj3 = (await usm.mintBurnAdjustment())
-            const decayFactor3 = wadDiv(ONE, EIGHT)
-            const targetMintBurnAdj3 = wadDecay(mintBurnAdj2, decayFactor3)
-
-            shouldEqualApprox(mintBurnAdj3, targetMintBurnAdj3, TEN)
+            const minFumBuyPrice5 = await usm.minFumBuyPrice()
+            const decayFactor5 = wadDiv(ONE, EIGHT)
+            const targetMinFumBuyPrice5 = wadMul(minFumBuyPrice4, decayFactor5)
+            shouldEqual(minFumBuyPrice5, targetMinFumBuyPrice5)
           })
 
-          it("allows burning FUM", async () => {
-            const targetEthPool = totalEthToFund.add(totalEthToMint) // Should be the total of ETH added: one fund() + one fund()
-            shouldEqual(ethPool, targetEthPool)
+          /* ____________________ Minting FUM (aka fund()), now at sliding price ____________________ */
+
+          describe("with FUM minted at sliding price", () => {
+            let ethPool3, debtRatio3, user2FumBalance3, totalFumSupply3, buySellAdj3, fumBuyPrice3, fumSellPrice3, usmBuyPrice3,
+                usmSellPrice3
+
+            beforeEach(async () => {
+              await usm.fund(user1, user2, ethPerFund, { from: user1 })
+
+              ethPool3 = await usm.ethPool()
+              debtRatio3 = await usm.debtRatio()
+              user2FumBalance3 = await fum.balanceOf(user2)
+              totalFumSupply3 = await fum.totalSupply()
+              buySellAdj3 = await usm.buySellAdjustment()
+              fumBuyPrice3 = await usm.fumPrice(sides.BUY)
+              fumSellPrice3 = await usm.fumPrice(sides.SELL)
+              usmBuyPrice3 = await usm.usmPrice(sides.BUY)
+              usmSellPrice3 = await usm.usmPrice(sides.SELL)
+            })
+
+            it("slides price correctly when minting FUM", async () => {
+              const targetEthPool3 = ethPool2.add(ethPerFund)
+              shouldEqual(ethPool3, targetEthPool3)
+
+              // Check vs the integral math in USM.fumFromFund():
+              const integralFirstPart = wadMul(debtRatio2, wadSquared(ethPool3).sub(wadSquared(ethPool2)))
+              const targetFumBalance3 = wadDiv(wadSqrt(integralFirstPart.add(wadSquared(wadMul(totalFumSupply1, fumBuyPrice2)))),
```

Comment:
If you use the same `wadSqrt` implementation in both the contract and the tests, then you should be sure beyond doubt the algorithm works. Are there tests for it anywhere?

Reply:
No, good point.  The code was lifted from production Uniswap (specifically from https://ethereum.stackexchange.com/a/87713) and I tested it pretty thoroughly, but only on an ad hoc basis - I'll add a proper test.

Reply:
Added.  (Have a look if you like, I snuck it into `03_USM.test.js`.)